src/presetsController.ts

```diff
@@ -816,11 +829,25 @@ export class PresetsController {
     }
 
     async getAllConfigurePresets(): Promise<preset.ConfigurePreset[]> {
-        preset.expandVendorForConfigurePresets(this.folderPath);
-        await preset.expandConditionsForPresets(this.folderPath, this._sourceDir);
         return preset.configurePresets(this.folderPath).concat(preset.userConfigurePresets(this.folderPath));
     }
 
+    async getAllBuildPresets(): Promise<preset.BuildPreset[]> {
+        return preset.buildPresets(this.folderPath).concat(preset.userBuildPresets(this.folderPath));
+    }
+
+    async getAllTestPresets(): Promise<preset.TestPreset[]> {
+        return preset.testPresets(this.folderPath).concat(preset.userTestPresets(this.folderPath));
+    }
+
+    async getAllPackagePresets(): Promise<preset.PackagePreset[]> {
```

Comment:
Are these used anywhere anymore?

Reply:
Yes, I didnt merge the two similar functions bc making everything async was a big change so im going to follow up with it